### PR TITLE
Run make check on OS X

### PR DESCRIPTION
**Pull request by ocefpaf**
We have

```
libgcc-4.8.5-1:
    libgcc_s.so.1 (lib/libgcc_s.so.1)
    libstdc++.so.6 (lib/libstdc++.so.6)
```

~~so I am adding `libgcc` here.~~

### CODE REVIEW

**Comment by ocefpaf** [comment]
OS X `make check` returns `FAIL: geos_unit`. Some googling suggested that `export CXXFLAGS="${CXXFLAGS} -stdlib=libc++ -std=c++11"` would fix it, but that was almost 4 years old post.

@jakirkham any ideas?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,10 +26,13 @@ requirements:
     build:
         - python  # [win]
         - msinttypes  # [win]
+    #run:
+        #- libgcc  # [unix]
```
Comment (jakirkham) [inline recipe/meta.yaml]: If we are not using Continuum's `gcc` on Linux, this can be dropped.

File: recipe/build.sh [hunk 2]
```diff
@@ -1,9 +1,22 @@
 #!/bin/bash
 
+if [[ `uname` == 'Darwin' ]]; then
+    export MACOSX_VERSION_MIN="10.7"
+    export CXXFLAGS="-mmacosx-version-min=${MACOSX_VERSION_MIN}"
+    export CXXFLAGS="${CXXFLAGS} -stdlib=libc++ -std=c++11"
+    export LINKFLAGS="-mmacosx-version-min=${MACOSX_VERSION_MIN}"
+    export LINKFLAGS="${LINKFLAGS} -stdlib=libc++ -std=c++11 -L${LIBRARY_PATH}"
```
Comment (jakirkham) [inline recipe/build.sh:8]: LGTM

**Comment by jakirkham** [comment]
It looks like there is a 3.5.0 version. Would you be willing to try that and see if doesn't fix the issue?

**Comment by jakirkham** [comment]
Did you see this [bug report](https://trac.osgeo.org/geos/ticket/299)? They have a patch to fix an issue in geos. Maybe also worth trying if upgrading is not an option.

File: recipe/meta.yaml (hunk 1 above)
Comment (ocefpaf) [inline recipe/meta.yaml]: I am never sure about this one. In this case the library links to `libgcc_s.so.1 (lib/libgcc_s.so.1)` and `libstdc++.so.6 (lib/libstdc++.so.6)` and those are provided by the `libgcc` package.

(See https://github.com/conda-forge/geos-feedstock/pull/10#issue-146018524)

**Comment by ocefpaf** [comment]
~~Let's try the upgrade first.~~

Edit: 3.5.0 broke Windows builds. I guess it is better to just wait for 3.5.1

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipe/meta.yaml]: Right, those will reliably be on the user's Linux system.

Comment (ocefpaf) [inline recipe/meta.yaml]: > Right, those will reliably be on the user's Linux system.

That means if we roll our own `libgcc` we should not add those then? (Sorry for the newbie questions.)

Comment (jakirkham) [inline recipe/meta.yaml]: No, that's a good question. IIRC we will include a `libgcc` package. However, that package will just copy over the libraries from the `gcc` that we build (and build other things with). See this PR ( https://github.com/conda-forge/staged-recipes/pull/135 ) for details about the proposal for doing this.

**Comment by jakirkham** [comment]
So, if you are still wanting to run `make check` on Mac, I would recommend trying to apply the [patch](https://trac.osgeo.org/geos/attachment/ticket/299/299-patch.patch) from that [bug report](https://trac.osgeo.org/geos/ticket/299). It looks like `geos` had a bug, which `clang` caught, but `gcc` had missed. The patch should make that test work properly and improve `geos` in the process.

**Comment by ocefpaf** [comment]
> So, if you are still wanting to run `make check` on Mac, I would recommend trying to apply the [patch](https://trac.osgeo.org/geos/attachment/ticket/299/299-patch.patch) from that [bug report](https://trac.osgeo.org/geos/ticket/299). It looks like `geos` had a bug, which `clang` caught, but `gcc` had missed. The patch should make that test work properly and improve `geos` in the process.

I did. See https://github.com/conda-forge/geos-feedstock/pull/10/files#diff-e178b687b10a71a3348107ae3154e44cR19

**Comment by jakirkham** [comment]
Oh sorry, I didn't realize you had and haven't had as much time to take a close look. So, that still didn't work. Will think about it after fixing `icu`.

**Comment by ocefpaf** [comment]
> Oh sorry, I didn't realize you had and haven't had as much time to take a close look. So, that still didn't work. Will think about it after fixing `icu`.

Don't worry too much about it. Looking at the mailing lists and trac I found out that this is an old (and recurring) issue. We are OK as long as we don't use that specific feature.

**Comment by ocefpaf** [comment]
This PR is stale and it is not worth salvaging it. We can re-do this later.

**Pull request closed without merging**